mantidproject/mantid
# Fix mantid py3 compatability warnings

**Description of work.**
This PR removes a couple of py3 deprecation warnings when running `launch_mantidworkbench.sh`:
1. Invalid escape sequence in `scriptcompatibility` regex
2. Deprecated sys method in `mainwindow` (workbench main script)

The remaining deprecation warnings are introduced by Mantid's dependencies. A separate issue has been created to stop posting these warnings to the logger #25216 

**To test:**
1. Run `launch_mantidworkbench.sh` in Py3
2. Of the deprecation warnings posted to the logger and the terminal, check that none of them are obviously Mantid-induced
3. Run workbench in py2 just to check that the sys compatibility method works

Fixes #25144 

*This does not require release notes* because **internal change**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
